ohshie.HabitTracker/DbOperations.cs

```diff
@@ -0,0 +1,263 @@
+            }
+            
+            connection.Close();
+
+            Console.WriteLine("######################");
+            foreach (var entry in habitData)
+            {
+                Console.WriteLine($"{entry.Id}. On {entry.Date}. You've read {entry.Quantity} {entry.AmountType}");
+            }
+            Console.WriteLine("######################");
+        }
+
+        return 1;
+    }
+
+    public void DeleteEntry(string dbName)
+    {
+        Console.Clear();
+        
+        PrintTable(dbName);
+        
+        using (var connection = new SqliteConnection(Dbconnection))
+        {
+            connection.Open();
+            
+            var tableCommand = connection.CreateCommand();
+            int invalidRow = 0;
+            int entryId = 0;
+            
+            while (invalidRow == 0)
+            {
+                entryId = Program.GetNumberFromUser("Id of entry you want to delete");
+                if (entryId == -1) return;
+
+                tableCommand.CommandText = $"delete from {dbName} where id = '{entryId}'";
+
+                invalidRow = tableCommand.ExecuteNonQuery();
+                if (invalidRow == 0) Console.WriteLine("Looks like there you've chosen entry that does not exist, or there are no entries left");
+            }
+            
+            connection.Close();
+            
+            Console.WriteLine($"Entry with Id {entryId} was deleted");
+        }
+    }
+
+    public void UpdateEntry(string dbName)
+    {
+        Console.Clear();
+        
+        using (var connection = new SqliteConnection(Dbconnection))
+        {
+            connection.Open();
+            var tableCommand = connection.CreateCommand();
+            int entryId = 0;
+            int checkEntryExist = 0;
+            
+            while (checkEntryExist == 0)
+            {
+                PrintTable(dbName);
+                entryId = Program.GetNumberFromUser("Id of entry you want to edit");
+                if (entryId == -1) return;
+                
+                tableCommand.CommandText = $"select exists(select 1 from {dbName} where Id = {entryId})";
+                
+                checkEntryExist = Convert.ToInt32(tableCommand.ExecuteScalar());
+
+                if (checkEntryExist == 0)
+                {
+                    Console.WriteLine("Looks like that entry does not exist!\n" +
+                                      "Press enter to try again");
+                    Console.ReadLine();
+                    Console.Clear();
+                }
+            }
+            
+            string newDate = Program.DateChooser();
+            if (newDate == "X") return;
+   
+            int newAmount = Program.GetNumberFromUser("new amount");
+            if (newAmount == -1) return;
+            
+            tableCommand.CommandText = $"update {dbName} set date = '{newDate}', quantity = {newAmount} WHERE Id = {entryId}";
+
+            tableCommand.ExecuteNonQuery();
```

Comment:
When updating a habit entry I'm getting this error: Are you currently able to update habits without issues?

![image](https://user-images.githubusercontent.com/108711476/228718842-3ff76e9e-3d4e-4da7-88af-f2bf02e2264f.png)